src/EditorFeatures/CSharpTest/EditAndContinue/RudeEditTopLevelTests.cs

```diff
@@ -6709,6 +6709,64 @@ public void TupleNames_TypeUpdate()
         }
 
         [Fact]
+        public void TupleElementUpdate()
+        {
+            var src1 = "class C { (int, int) M() { return (1, 2); } }";
+            var src2 = "class C { (int, int) M() { return (1, 7); } }";
+
+            var edits = GetTopEdits(src1, src2);
+
+            edits.VerifyEdits(
+                "Update [(int, int) M() { return (1, 2); }]@10 -> [(int, int) M() { return (1, 7); }]@10");
+
+            edits.VerifyRudeDiagnostics();
+        }
+
+        [Fact]
+        public void TupleElementDelete()
+        {
+            var src1 = "class C { (int, int, int a) M() { return (1, 2, 3); } }";
+            var src2 = "class C { (int, int) M() { return (1, 2); } }";
+
+            var edits = GetTopEdits(src1, src2);
+
+            edits.VerifyEdits(
+                "Update [(int, int, int a) M() { return (1, 2, 3); }]@10 -> [(int, int) M() { return (1, 2); }]@10");
+
+            edits.VerifyRudeDiagnostics(
+                Diagnostic(RudeEditKind.TypeUpdate, "(int, int) M()", "method"));
+        }
+
+        [Fact]
+        public void TupleElementAdd()
+        {
+            var src1 = "class C { (int, int) M() { return (1, 2); } }";
+            var src2 = "class C { (int, int, int a) M() { return (1, 2, 3); } }";
+
+            var edits = GetTopEdits(src1, src2);
+
+            edits.VerifyEdits(
+                "Update [(int, int) M() { return (1, 2); }]@10 -> [(int, int, int a) M() { return (1, 2, 3); }]@10");
+
+            edits.VerifyRudeDiagnostics(
+                Diagnostic(RudeEditKind.TypeUpdate, "(int, int, int a) M()", "method"));
+        }
+
+        [Fact]
+        public void ParenthesizedVariables_NamesUpdate()
```

Comment:
>ParenthesizedVariables_NamesUpdate [](start = 20, length = 34)

This is really not testing much since the update is in a method body, which is basically ignored when analyzing top-level changes.

It'd be better to test name change in the return type, e.g.

(int a, int b) M() ... 

to 

(int a, int c) M() ... 

it should be a rude edit to make such change